code/modules/admin/verbs/deathsquad.dm

```diff
@@ -0,0 +1,192 @@
+	commando_number = clamp(input(src, "How many Deathsquad Commandos would you like to send? (Recommended is 6, Max is [MAX_COMMANDOS]])", "Specify Commandos") as num|null, 1, MAX_COMMANDOS)
+	if(!commando_number)
+		message_admins("[key_name_admin(proccaller)] cancelled their Deathsquad.")
+		log_admin("[key_name(proccaller)] cancelled their Deathsquad.")
+		return
+	if(GLOB.deathsquad_sent)
+		if(alert("A Deathsquad leader has previously been sent with an unrestricted NAD, would you like to spawn another?",, "Yes", "No") != "Yes")
+			is_leader = FALSE
+	GLOB.deathsquad_sent = TRUE
+	message_admins("[key_name_admin(proccaller)] has sent a Deathsquad with [commando_number] commandos.")
+	log_admin("[key_name(proccaller)] has sent a Deathsquad with [commando_number] commandos.")
+
+	// Find the nuclear auth code
+	var/nuke_code
+	var/temp_code
+	for(var/obj/machinery/nuclearbomb/N in GLOB.machines)
+		temp_code = text2num(N.r_code)
+		if(temp_code)//if it's actually a number. It won't convert any non-numericals.
+			nuke_code = N.r_code
+			break
+
+	// Find ghosts willing to be DS
+	var/list/commando_ghosts = list()
+	if(alert("Would you like to custom pick your Deathsquad?",, "Yes", "No") == "Yes")
+		var/image/source = image('icons/obj/cardboard_cutout.dmi', "cutout_deathsquad")
+		commando_ghosts = pollCandidatesWithVeto(src, usr, commando_number, "Join the DeathSquad?",, 21, 60 SECONDS, TRUE, GLOB.role_playtime_requirements[ROLE_DEATHSQUAD], TRUE, FALSE, source = source)
+	else
+		var/image/source = image('icons/obj/cardboard_cutout.dmi', "cutout_deathsquad")
+		commando_ghosts = SSghost_spawns.poll_candidates("Join the Deathsquad?",, GLOB.responseteam_age, 60 SECONDS, TRUE, GLOB.role_playtime_requirements[ROLE_DEATHSQUAD], TRUE, FALSE, source = source)
+		if(length(commando_ghosts) > commando_number)
+			commando_ghosts.Cut(commando_number + 1) //cuts the ghost candidates down to the amount requested
+	if(!length(commando_ghosts))
+		message_admins("[key_name_admin(proccaller)]'s Deathsquad had no volunteers and was cancelled.")
+		log_admin("[key_name(proccaller)]'s Deathsquad had no volunteers and was cancelled.")
+		to_chat(src, "<span class='userdanger'>Nobody volunteered to join the DeathSquad.</span>")
+		return
+
+	// Spawns commandos and equips them.
+	for(var/obj/effect/landmark/spawner/ds/L in GLOB.landmarks_list) //Despite obj/effect/landmark/spawner/ds being in the exact same location and doing the exact same thing as obj/effect/landmark/spawner/ert, switching them breaks it.
+		if(!commando_number)
+			break
+		if(!length(commando_ghosts))
+			break
+		var/use_ds_borg = FALSE
+		var/mob/ghost_mob = pick(commando_ghosts)
+		commando_ghosts -= ghost_mob
+		if(!ghost_mob || !ghost_mob.key || !ghost_mob.client)
+			continue
+
+		if(!is_leader)
+			var/new_dstype = alert(ghost_mob.client, "Select Deathsquad Type.", "Deathsquad Character Generation", "Organic", "Cyborg")
+			if(new_dstype == "Cyborg")
+				use_ds_borg = TRUE
```

Comment:
Hmm.. this means that if someone signs up, but then never clicks the button "would you like to be a borg" button, then we'd get stuck here, and never advance to polling other people/assembling the team..

One way to work around this is putting all this, and the rest of mob creation code below in a dedication function, which you invoke via a `spawn()` similar.